tests/ui/const-generics/adt_const_params/mismatch-raw-ptr-in-adt.rs

```diff
@@ -0,0 +1,13 @@
+//! Regression test for <https://github.com/rust-lang/rust/issues/131052>
```

Comment:
I think what's happened here is that we're now masking an ICE that was originally caused by `lit_to_const` returning `ValTree`'s with incorrect types :thinking: I can't think of a way to write a test that exposes that underlying brokenness so I think this is fine.

It would be nice to fix the underlying brokenness regardless. Would you be interested in doing that? The short explanation is that we `lit_to_const` for byte strings doesn't check that the type of the parameter the byte string is an argument to, is `&[u8]` is only checks if its a reference to a slice.